pkg/policy/validate.go

```diff
@@ -0,0 +1,112 @@
+// Copyright 2023 The Sigstore Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package policy
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+
+	"github.com/sigstore/policy-controller/pkg/apis/policy/v1alpha1"
+	"github.com/sigstore/policy-controller/pkg/apis/policy/v1beta1"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"knative.dev/pkg/apis"
+)
+
+var (
+	// ErrEmptyDocument is the error returned when no document body is
+	// specified.
+	ErrEmptyDocument = errors.New("document is required to create policy")
+
+	// ErrUnknownType is the error returned when a type contained in the policy
+	// is unrecognized.
+	ErrUnknownType = errors.New("unknown type")
+)
+
+// Validate decodes a provided YAML document containing zero or more objects
+// and performs limited validation on them.
```

Comment:
Might want to also document that this defaults the fields for the CIPs? It's not obvious since it's buried in validate function, and I was looking for it. Others using it might wonder about it too :)

Reply:
Will add to next push.  thanks!